tests/test_commands.py

```diff
@@ -109,3 +109,46 @@ def test_enqueue_files(self):
         self.__send("enqueue-files "
                     "one,two\\, please,slash\\\\.mp3,four")
         self.assertEquals(app.window.playlist.q.get(), songs)
+
+    def test_with_pattern(self):
+        songs = [AudioFile({"~filename": fn, "title": fn.upper()})
+                 for fn in ["one", "two, please", "slash\\.mp3", "4.0-four"]]
+        app.library.add(songs)
+
+        self.assertFalse(app.window.playlist.q.get())
+        self.__send("enqueue-files "
+                    "one,two\\, please,slash\\\\.mp3,4.0-four")
+        # Old syntax
+        assert self.__send("print-query two ") == "two, please\n"
+        # Old syntax that is also a valid json
```

Comment:
This is all much better :+1:

Could you split the assertions up into tests? Motivation: it's way easier to debug when a test called `test_old_syntax_that_is_valid_json()` starts failing, than "line 123 of _test_everything()`" etc

The comments pretty much become the test name, and anything shared is a fixture (ideally) or here perhaps just in `setUp()`

Reply:
I refactored these tests into a new class as requested.  (I think the resulting code is a bit complex.)  Thanks.

Reply:
Great, thanks!